include/dslash_quda.h

```diff
@@ -475,11 +475,35 @@ namespace quda {
      @param[in] comm_override Override for which dimensions are partitioned
      @param[in] profile The TimeProfile used for profiling the dslash
   */
+
   void ApplyDomainWall4D(ColorSpinorField &out, const ColorSpinorField &in, const GaugeField &U, double a, double m_5,
                          const Complex *b_5, const Complex *c_5, const ColorSpinorField &x, int parity, bool dagger,
                          const int *comm_override, TimeProfile &profile);
 
-  enum Dslash5Type { DSLASH5_DWF, DSLASH5_MOBIUS_PRE, DSLASH5_MOBIUS, M5_INV_DWF, M5_INV_MOBIUS, M5_INV_ZMOBIUS };
+  enum Dslash5Type {
+    DSLASH5_DWF,
+    DSLASH5_MOBIUS_PRE,
+    DSLASH5_MOBIUS,
+    M5_INV_DWF,
+    M5_INV_MOBIUS,
+    M5_INV_ZMOBIUS,
+    M5_EOFA,
+    M5INV_EOFA
+  };
+
+  enum MdwfFusedDslashType {
+    /**
+      Applying the following five kernels in the order of 4-0-1-2-3 is equivalent to applying
+      the full even-odd preconditioned symmetric MdagM operator:
+      op = (1 - M5inv * D4 * D5pre * M5inv * D4 * D5pre)^dag
+            * (1 - M5inv * D4 * D5pre * M5inv * D4 * D5pre)
+    */
+    dslash4_dslash5pre_dslash5inv = 0,
```

Comment:
QUDA convention, `enum` (or `enum class`, see above) should be all capitals. Comment should go outside the `enum`.